warehouse/migrations/versions/be62a4cd76e3_add_a_column_to_store_the_entire_set_of_.py

```diff
@@ -0,0 +1,66 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Add a column to store the entire set of serialized caveats
+
+Revision ID: be62a4cd76e3
+Revises: 812e14a4cddf
+Create Date: 2024-01-20 16:28:31.573452
+"""
+
+import sqlalchemy as sa
+
+from alembic import op
+from sqlalchemy.dialects import postgresql
+
+revision = "be62a4cd76e3"
+down_revision = "812e14a4cddf"
+
+
+def upgrade():
+    op.add_column(
+        "macaroons",
+        sa.Column("caveats", postgresql.JSONB(astext_type=sa.Text()), nullable=True),
+    )
+
+    # Where our permissions_caveat is {"permission": "user"}, set our caveats to
+    # [[3, str(user_id)]], which is a single RequestUser caveat where the user_id
+    # is the attached user_id for this Macaroon.
+    op.execute(
+        """ UPDATE macaroons
+            SET caveats = jsonb_build_array(jsonb_build_array(3, user_id::text))
+            WHERE
+                caveats IS NULL
+                AND user_id IS NOT NULL
+                AND permissions_caveat->>'permissions' = 'user'
+        """
+    )
```

Comment:
This is going to hit some ~80k rows in production.
For a SELECT to find them, the planner chose to do a Parallel Seq Scan that takes ~320ms to select (without the new `caveats` column, since it doesn't exist yet).

```sql
warehouse=> explain analyze select * from macaroons where user_id is not null AND permissions_caveat->>'permissions' = 'user';
                                                          QUERY PLAN
-------------------------------------------------------------------------------------------------------------------------------
 Gather  (cost=1000.00..7224.11 rows=725 width=222) (actual time=1.364..313.752 rows=88549 loops=1)
   Workers Planned: 2
   Workers Launched: 2
   ->  Parallel Seq Scan on macaroons  (cost=0.00..6151.61 rows=302 width=222) (actual time=0.414..297.411 rows=29516 loops=3)
         Filter: ((user_id IS NOT NULL) AND ((permissions_caveat ->> 'permissions'::text) = 'user'::text))
         Rows Removed by Filter: 30226
 Planning Time: 0.497 ms
 Execution Time: 319.197 ms
(8 rows)
```

I suspect that both the UPDATEs will happen in under our statement timeout of 4 seconds, so I wanted to note the timing I saw in case we hit a migration problem.